src/main/java/seedu/address/model/student/YearMatchPredicate.java

```diff
@@ -1,30 +1,34 @@
 package seedu.address.model.student;
 
+import java.util.List;
 import java.util.function.Predicate;
 
+import seedu.address.commons.util.StringUtil;
+
 /**
  * Tests that a {@code Student}'s {@code Year} matches the given year.
  */
 public class YearMatchPredicate implements Predicate<Student> {
 
     // Attributes
-    private final Year year;
+    public final List<String> keywords;
 
     // Constructor
-    public YearMatchPredicate(Year year) {
-        this.year = year;
+    public YearMatchPredicate(List<String> keywords) {
+        this.keywords = keywords;
     }
 
     @Override
     public boolean test(Student student) {
-        return this.year.equals(student.getYear());
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(student.getYear().year, keyword));
```

Comment:
does this mean that if you search "sec 3" you get all that either match "sec" or match "3"? (e.g. you get sec 1-4 and p3, s3)

Reply:
Yep something similar currently "sec 3" would match "sec 1-4", "p 3" and "s 3" but not "p3" or "s3" as those it matches by whole words so "p3" is not "sec" or "3" hence it won't match. Same for "s3"!

Reply:
Would it be more helpful if we made it look only for words that contain all the keywords (even incomplete ones)? (i.e. `AND` search)

For example it would be like:
`String year = student.getYear().year.toLowerCase();`
`return keywords.stream().map(String::toLowerCase).allMatch(year::contains);`